docs/scos/dev/tutorials-and-howtos/howtos/howto-reduce-jenkins-execution-costs-without-refactoring.md

```diff
@@ -0,0 +1,483 @@
+     */
+    protected QueueClientInterface $queueClient;
+
+    /**
+     * @var array<string>
+     */
+    protected array $queueNames;
+
+    /**
+     * @var \Spryker\Zed\Queue\Business\SignalHandler\SignalDispatcherInterface
+     */
+    protected SignalDispatcherInterface $signalDispatcher;
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    protected LoggerInterface $logger;
+
+    /**
+     * @var \Pyz\Zed\Queue\Business\Strategy\QueueProcessingStrategyInterface
+     */
+    protected QueueProcessingStrategyInterface $queueProcessingStrategy;
+
+    /**
+     * @var \SplFixedArray<\Symfony\Component\Process\Process>
+     */
+    protected SplFixedArray $processes;
+
+    /**
+     * @var int
+     */
+    private int $runningProcessesCount = 0;
+
+    /**
+     * @var \Pyz\Zed\Queue\Business\Worker\WorkerStats
+     */
+    private WorkerStats $stats;
+
+    /**
+     * @var \Pyz\Zed\Queue\Business\SystemResources\SystemResourcesManagerInterface
+     */
+    private SystemResourcesManagerInterface $sysResManager;
+
+    /**
+     * @var array<string, float>
+     */
+    private array $timers = [];
+
+    public function __construct(
+        ProcessManagerInterface $processManager,
+        QueueConfig $queueConfig,
+        QueueClientInterface $queueClient,
+        array $queueNames,
+        QueueProcessingStrategyInterface $queueProcessingStrategy,
+        SignalDispatcherInterface $signalDispatcher,
+        SystemResourcesManagerInterface $sysResManager,
+        LoggerInterface $logger
+    ) {
+        $this->processManager = $processManager;
+        $this->queueConfig = $queueConfig;
+        $this->queueClient = $queueClient;
+        $this->queueNames = $queueNames;
+        $this->queueProcessingStrategy = $queueProcessingStrategy;
+        $this->signalDispatcher = $signalDispatcher;
+        $this->signalDispatcher->dispatch($this->queueConfig->getSignalsForGracefulWorkerShutdown());
+        $this->sysResManager = $sysResManager;
+        $this->logger = $logger;
+
+        $this->processes = new SplFixedArray($this->queueConfig->getQueueWorkerMaxProcesses());
+        $this->stats = new WorkerStats();
+    }
+
+    /**
+     * @param string $timerName
+     * @param string $message
+     * @param string $level
+     * @param int $intervalSec
+     *
+     * @return void
+     */
+    protected function logNotOftenThan(string $timerName, string $message, string $level = 'debug', int $intervalSec = 1): void
+    {
+        if (microtime(true) - ($this->timers[$timerName] ?? 0) >= $intervalSec) {
+            $this->timers[$timerName] = microtime(true);
+            $this->logger->$level($message);
+        }
+    }
+
+    /**
+     * @param string $command
+     * @param array<string, mixed> $options
+     *
+     * @return void
+     */
+    public function start(string $command, array $options = []): void
+    {
+        $maxThreshold = $this->queueConfig->getQueueWorkerMaxThreshold();
```

Comment:
just from the readability point of view, could you describe this config values here instead of in the bottom of the page. It's pretty clear how to "connect" them, but it requires a bit more time to do so